proposals/Hole Punching.md

```diff
@@ -0,0 +1,146 @@
+*   Dogfood an alpha version of Hole Punching to Labbers and collect metrics regarding success rates and resource consumption.
+*   Based on the results of the dog-fooding endeavour, fix/optimize the hole punching process and release it to the world in a phased manner.
+*   In the initial phase,  ONLY use statically configured Limited Relay servers hosted by PL.
+*   Once we conclude that PL hosted Limited Relays are stable,  ship a release that turns on the Limited Relay protocol in public DHT servers but continues to use the statically configured Limited Relays.
+*   Once ~30% of public DHT servers upgrade to support the Limited Relay protocol(measure using Hydra Boosters), ship automated discovery & use of Limited Relays to coordinate a hole-punch rather than using statically configured Limited Relays servers.
+*   Achieve ~90% hole punching success if both peers are behind a Cone NAT  in the second dog-fooding phase that uses AutoRelay to discover and connect to  Limited Relays in the wild.
+*   Massive focus on usability, user education and evangelizing the feature by writing blog posts discussing a “how we got here”/ internals / engineering journey of Hole Punching in libp2p with metrics and blog posts and demo videos on how to configure, use and debug Hole Punching.
+
+### Technical deliverables
+
+*   **Phase1**
+    - ~90% hole punching success if both peers are behind a Cone NAT in the first dog-fooding phase using PL hosted Limited Relays.
+*   **Phase 2**
+    - Optimise based on Dogfooding results and metrics and ship the feature using statically configured PL hosted Limited Relays.
+*   **Phase 3**
+    - Once we conclude that PL hosted Limited Relays are stable,  ship a release that turns on the Limited Relay protocol in public DHT servers.
+*   **Phase 4**   
+    - Once ~30% of public DHT servers upgrade to support the Limited Relay protocol (measure using Hydra Boosters), ship automated discovery & use of Limited Relays to coordinate a hole-punch rather than using statically configured Limited Relays servers.
+*   **Phase 5**
+    - ~90% hole punching success if both peers are behind a Cone NAT  in the second dog-fooding phase that uses AutoRelay to discover and connect to Limited Relays in the wild rather than using statically configured Limited Relays.
+
+### Success criteria
+
+*   Dog-fooding phases deliver ~90% success for labbers using Home ISPs with Cone NATs (TCP & QUIC).
+*   No bugs related to Hole Punching failures if both peers involved in a Hole Punch are behind a Cone NAT (we have good PRs for and will ship code/tools for users to detect their NAT type).
+*   Users do not file bug reports about their public DHT peers getting DDosed/consuming too much bandwidth/resources because of acting as Limited Relays.
+*   We receive great traction and feedback on the ease of use and robustness of Hole Punching on channels such as Twitter, user surveys and from our community of users/partners.
+
```

Comment:
I would suggest to add a criteria such as "IPFS node is able to tell if it can be dialed (hole punched to), combined with some troubleshooting interface that can guide a node operator in terms of what to do to make node reachable"